custom_components/kia_uvo/button.py

```diff
@@ -0,0 +1,215 @@
+        return True
+    if getattr(v, "ev_charge_limits_dc", None) is not None:
+        return True
+
+    return False
+
+
+BUTTONS: list[KiaButtonDescription] = [
+    KiaButtonDescription(
+        key="update_cached",
+        name="Update (cached)",
+        icon="mdi:update",
+        service="update",
+    ),
+    KiaButtonDescription(
+        key="force_update",
+        name="Force update",
+        icon="mdi:cloud-refresh",
+        service="force_update",
+    ),
+    KiaButtonDescription(
+        key="lock",
+        name="Lock",
+        icon="mdi:lock",
+        service="lock",
+    ),
+    KiaButtonDescription(
+        key="unlock",
+        name="Unlock",
+        icon="mdi:lock-open",
+        service="unlock",
+    ),
+    KiaButtonDescription(
+        key="start_climate",
+        name="Start climate",
+        icon="mdi:car-seat-heater",
+        service="start_climate",
+    ),
+    KiaButtonDescription(
+        key="stop_climate",
+        name="Stop climate",
+        icon="mdi:car-seat-cooler",
+        service="stop_climate",
+    ),
+    KiaButtonDescription(
+        key="start_charge",
+        name="Start charge",
+        icon="mdi:ev-station",
+        service="start_charge",
+        capability=_is_ev,
+    ),
+    KiaButtonDescription(
+        key="stop_charge",
+        name="Stop charge",
+        icon="mdi:ev-station-off",
+        service="stop_charge",
+        capability=_is_ev,
+    ),
+    KiaButtonDescription(
+        key="open_charge_port",
+        name="Open charge port",
+        icon="mdi:ev-plug-type2",
+        service="open_charge_port",
+        capability=_is_ev,
+    ),
+    KiaButtonDescription(
+        key="close_charge_port",
+        name="Close charge port",
+        icon="mdi:ev-plug-ccs2",
+        service="close_charge_port",
+        capability=_is_ev,
+    ),
+    KiaButtonDescription(
+        key="set_charge_limits_80",
+        name="Set AC/DC limit to 80%",
+        icon="mdi:battery-80",
+        service="set_charge_limits",
+        payload={"ac_limit": 80, "dc_limit": 80},
+        capability=_is_ev,
+    ),
+    KiaButtonDescription(
+        key="set_charge_limits_100",
+        name="Set AC/DC limit to 100%",
+        icon="mdi:battery",
+        service="set_charge_limits",
+        payload={"ac_limit": 100, "dc_limit": 100},
+        capability=_is_ev,
+    ),
+    KiaButtonDescription(
+        key="set_charging_current_100",
+        name="Charging current 100%",
+        icon="mdi:current-ac",
+        service="set_charging_current",
+        payload={"level": 100},
+        capability=_is_ev,
+    ),
+    KiaButtonDescription(
+        key="set_charging_current_60",
+        name="Charging current 60%",
+        icon="mdi:current-ac",
+        service="set_charging_current",
+        payload={"level": 60},
+        capability=_is_ev,
+    ),
+    KiaButtonDescription(
+        key="start_hazard_lights",
```

Comment:
I think this plus the ones below are only newer cars. Not sure if we have a way to check for that though.   Thoughts?